lib/ide/src/transport/web.rs

```diff
@@ -0,0 +1,250 @@
+//! web_sys::WebSocket-based `Transport` implementation.
+
+use crate::prelude::*;
+
+use basegl_system_web::closure::ClosureStorage;
+use failure::Error;
+use futures::channel::mpsc;
+use json_rpc::Transport;
+use json_rpc::TransportEvent;
+use web_sys::CloseEvent;
+use web_sys::Event;
+use web_sys::MessageEvent;
+
+
+
+// =================
+// === Utilities ===
+// =================
+
+#[wasm_bindgen]
+extern "C" {
+    /// Converts given `JsValue` into a `String`.
+    #[allow(unsafe_code)]
+    #[wasm_bindgen(js_name="String")]
+    fn js_to_string(s: JsValue) -> String;
+}
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+/// Errors that may happen when trying to establish WebSocket connection.
+#[derive(Clone,Debug,Fail)]
+pub enum ConnectingError {
+    /// Failed to construct websocket. Usually this happens due to bad URL.
+    #[fail(display = "Invalid websocket specification: {}.", _0)]
+    ConstructionError(String),
+    /// Failed to establish connection. Usually due to connectivity issues,
+    /// wrong URL or server being down. Unfortunately, while the real error
+    /// cause is usually logged down in js console, we have no reliable means of
+    /// obtaining it programmatically. Reported error codes are utterly
+    /// unreliable.
```

Comment:
Is there absolutely no way to even parse the error and get this information ? :(

Reply:
None that I know of. Nor the people on Stack Overflow. 
Reportedly they make it impossible for security reasons.